packages/mui-base/src/useSlider/useSlider.ts

```diff
@@ -267,7 +267,7 @@ export function useSlider(parameters: UseSliderParameters): UseSliderReturnValue
 
   const [focusedThumbIndex, setFocusedThumbIndex] = React.useState(-1);
 
-  const sliderRef = React.useRef<HTMLSpanElement>();
+  const sliderRef = React.useRef<HTMLSpanElement>(undefined);
```

Comment:
Shouldn't it be?

```suggestion
  const sliderRef = React.useRef<HTMLSpanElement>(null);
```

It should be an immutable ref here. It looks like I have missed this one in #38762. We could make a PR just for it.

I dove a bit into this in https://github.com/mui/mui-x/pull/11847.

Reply:
Oh, ok, it's changing: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64896, so 

```suggestion
  const sliderRef = React.useRef<HTMLSpanElement | null>(null);
```

?

Reply:
I think `| null` is the correct approach, yeah https://github.com/mui/material-ui/pull/42881#discussion_r1668142722

Reply:
Right, then +1 to udpate all the codebase with this. We don't have to correlate it with React 19..it seems that it would work with all versions.

Reply:
For posterity, in React 19:
- `React.useRef` requires an initial value. There's no need to do `| null` when the initial value is `null` (same for `undefined`). You can still do it if you want to be more explicit. It goes like this (pseudo-code):
  - `const ref = React.useRef<T>(null)` -> `ref` is `T | null`
  - `const ref = React.useRef<T>(undefined)` -> `ref` is `T | undefined`
  - `const ref = React.useRef<T>(T)` -> `ref` is `T`
- `React.MutableRefObject` has been deprecated. `React.RefObject` is used for all cases now, and it's always mutable. Unlike `React.Ref`, we need to explicitly make the ref object nullable: `React.RefObject<T | null>`.

(more info in the React 19 types [PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69022/files#diff-32cfd8cb197872bcba371f5018185d2e75fa540b52cda2dd7d8ac12dcc021299))